src/foam/u2/dialog/Popup.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * @license
+ * Copyright 2014 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+foam.CLASS({
+  package: 'foam.u2.dialog',
+  name: 'Popup',
+  extends: 'foam.u2.Element',
+
+  imports: [
+    'document'
+  ],
+  exports: [
+    'as overlay'
+  ],
+
+  axioms: [
+    foam.u2.CSS.create({
+      code: function() {/*
+        ^ {
+          align-items: center;
+          bottom: 0;
+          display: flex;
+          justify-content: space-around;
+          left: 0;
+          position: fixed;
+          right: 0;
+          top: 0;
+          z-index: 1000;
+        }
+        ^container {
+          align-items: center;
+          display: flex;
+          height: 100%;
+          justify-content: space-around;
+          position: relative;
+          width: 100%;
+        }
+        ^background {
+          background-color: #000;
+          bottom: 0;
+          left: 0;
+          opacity: 0.4;
+          position: absolute;
+          right: 0;
+          top: 0;
+        }
+        ^inner {
+          z-index: 3;
+        }
+      */}
+    })
+  ],
+
+  properties: [
+    [ 'backgroundColor', '#fff' ]
+  ],
+
+  methods: [
+    function init() {
```

Comment:
I think a more functional style would be easier to read here.  You can set the content by doing .start(/*container*/).call(function() { self.container = this })..., which wouldn't require you to use local variable and break out of one continuous statement.